dissect/target/plugins/os/windows/regf/usb.py

```diff
@@ -35,127 +44,173 @@
+                        friendly_name=friendly_name,
                         serial=serial,
-                        vid=None,
-                        pid=None,
-                        vendor=device_info["vendor"],
-                        product=device_info["product"],
-                        version=device_info["version"],
-                        containerid=containerid,
-                        first_install=timestamps["first_install"],
-                        first_insert=timestamps["first_insert"],
-                        last_insert=timestamps["last_insert"],  # AKA first arrival
-                        last_removal=timestamps["last_removal"],
-                        info_origin=info_origin,
+                        container_id=container_id,
+                        **device_info,
+                        **timestamps,
+                        volumes=list(self.find_volumes(serial)),
+                        mounts=mounts,
+                        users=[
+                            u.user.name
+                            for u in self.find_users([m[10:] for m in mounts if m.startswith("\\??\\Volume{")])
+                        ],
+                        source=self.USB_STOR,
                         _target=self.target,
                     )
+
+    def find_volumes(self, serial: str) -> Iterator[str]:
+        """Attempts to find mounted volume names for the given serial"""
+
+        try:
+            for device in self.target.registry.key(self.PORTABLE_DEVICES).subkeys():
+                if serial in device.name.lower():
+                    yield device.value("FriendlyName").value
+        except RegistryKeyNotFoundError:
+            pass
+
+    def find_mounts(self, serial: str) -> Iterator[str]:
+        """Attempts to find drive letters the given serial has been mounted on"""
+
+        try:
+            for mount in self.target.registry.key(self.MOUNT_LETTER_MAP).values():
+                try:
+                    if serial in mount.value.decode("utf-16-le").lower():
+                        yield mount.name.replace("\\DosDevices\\", "")
+                except UnicodeDecodeError:
+                    pass
+        except RegistryKeyNotFoundError:
+            pass
+
+    def find_users(self, volume_guids: list[str]) -> Iterator[str]:
+        """Attempt to find Windows users that have interacted with the given volume guid(s)"""
+
+        for volume_guid in volume_guids:
+            try:
+                for key in self.target.registry.key(self.USER_MOUNTS + "\\" + volume_guid):
+                    yield self.target.registry.get_user_details(key)
+            except RegistryKeyNotFoundError:
+                pass
+
+
+def unpack_timestamps(usb_reg_properties: VirtualKey) -> dict[str, int]:
+    """Unpack relevant Windows timestamps from the provided USB registry properties subkey.
+
+    Args:
+        usb_reg_properties (VirtualKey): A registry object with USB properties
```

Comment:
```suggestion
        usb_reg_properties: A registry object with USB properties.
```
Type info is already in the function signature.